apps/backend/src/modules/points-of-interest/points-of-interest.service.ts

```diff
@@ -47,15 +48,28 @@ class PointsOfInterestService implements Service {
 		return true;
 	}
 
-	public async findAll(): Promise<
-		CollectionResult<PointsOfInterestResponseDto>
-	> {
-		const items = await this.pointsOfInterestRepository.findAll();
+	public async findAll(
+		query: null | PointsOfInterestSearchQuery,
+	): Promise<CollectionResult<PointsOfInterestResponseDto>> {
+		const DEFAULT_SEARCH_RADIUS_KM = 5;
+		const THOUSAND = 1000;
+
+		const hasLocationFilter =
+			Boolean(query?.latitude) && Boolean(query?.longitude);
+
+		const radiusMeters = (query?.radius ?? DEFAULT_SEARCH_RADIUS_KM) * THOUSAND;
+
+		const searchParameters = {
+			...query,
+			radius: radiusMeters,
+		} as PointsOfInterestSearchQuery;
```

Comment:
why do we need type cast here, shouldn't it match the type
```suggestion
		const searchParameters: PointsOfInterestSearchQuery = {
			...query,
			radius: radiusMeters,
		};
```

Reply:
because we don't know if there are cords or not

<img width="701" height="475" alt="image" src="https://github.com/user-attachments/assets/52a698d5-662d-4013-bc03-21a976350558" />